sycl/doc/extensions/PropertyList/SYCL_EXT_ONEAPI_properties.asciidoc

```diff
@@ -0,0 +1,682 @@
+:sectnums:
+
+= `SYCL_EXT_ONEAPI_PROPERTIES`
+
+== Introduction
+
+The `sycl::property_list` found in SYCL 2020 is used to store properties used in the construction of runtime classes. It does so in a fully dynamic manner, such that it is not possible to obtain any useful information about the types of properties passed nor their values at compile time.
+
+Compile-time-constant properties are an important building block for classes and functions that have a need to propagate compile-time information for semantic and optimization purposes, while runtime properties continue to serve an important role in dynamic parameter specification.
+
+This extension introduces `sycl::ext::oneapi::properties`, which is a replacement for `sycl::property_list` that supports the storage and manipulation of compile-time-constant properties in addition to runtime properties.
```

Comment:
Sorry, I thought of something else ... didn't we decide that the first implementation will be experimental?  If so, we should change the namespace to `sycl::ext::oneapi::experimental::properties`.

Tagging @steffenlarsen to see if this matches the implementation he is working on.

Reply:
> Sorry, I thought of something else ... didn't we decide that the first implementation will be experimental? If so, we should change the namespace to sycl::ext::oneapi::experimental::properties.

Yes, I believe we agreed that it should be experimental for now. Whether or not the namespace change should be done in this PR I'll let you decide.